### DISCUSSION TITLE
Ez -Insar first impression

### DISCUSSION THREAD

**Discussion by Pierparide** [alexisInSAR/EZ-InSAR#24] [Help]
i’ve noted that when import a kml file in the Ez-Insar appear a window in which i don’t see any study area and i can not manually select the same. There’s a command to select area?

**Comment by alexisInSAR** [comment]
Hi Pierparide, 

At the beginning of processing, you provide a .kml file (polygon format) to generate two other .kml files: “target.kml” and “area.kml”. “target.kml” is the same file provided by the user and stored in the work directory. It is used to keep the extent of the Region of Interest (ROI).

“area.kml” is the extent of the ROI defined by a rectangle polygon. In this case, you can manually modify “area.kml” and "target.get" using a text editor or GIS software (i.e., QGIS, Google Earth, etc.) as EZ-InSAR extracts the geographical coordinates from this file before each processing. If the modified files are not a rectangle, EZ-InSAR should use the maximum and minimum values of longitude/latitude from this file.

After the selection of the .kml file, a map (see image) must be visible. 

Alexis 

<img width="862" alt="image" src="https://user-images.githubusercontent.com/107697895/225444933-dea6ca18-0e45-422e-b3f2-7840c772f400.png">

**Reply by Pierparide** [reply]
Ok….now is ok……..follow question is Why chek slcs list process is too long for only 1 year of sentinel 1A mode IW?Depends on what?

**Reply by alexisInSAR** [reply]
Checking the available SLCs (the creation of the lists) takes a few seconds. Please check your terminal to ensure that there are no issues and that the latency is related to the ASF server.

**Reply by Pierparide** [reply]
I obtain this errors….how resolve it
![Uploading 68E2EB4A-F117-4F2C-AF27-F91984E8F80A.jpeg…]()

**Reply by Pierparide** [reply]
![ACB93D4A-8C3C-47FD-997B-D2B94933A134](https://user-images.githubusercontent.com/94765448/225534429-77863927-ec70-48a9-b07a-88a1c88a10a9.jpeg)

**Reply by alexisInSAR** [reply]
Can you test the creation of the list with the new version of the script named "createlistSLC.m"?

**Reply by Pierparide** [reply]
What do you mean for “new version of the script”? I’ve 2.0.0 script version…..there is a new update? Is it possible to replace it?thank you

**Reply by alexisInSAR** [reply]
Regarding the updates, only the "createlistSLC.m" has been modified. You can directly copy and paste the script into your EZ-InSAR directory. 

#23

**Reply by Pierparide** [reply]
![image](https://user-images.githubusercontent.com/94765448/225657262-45e58d2a-057c-489a-8dcb-37d87d031960.jpg)

**Reply by Pierparide** [reply]
what is this error due to?
![image](https://user-images.githubusercontent.com/94765448/225658280-0107c223-4d28-4bb4-852e-4711f28efdc9.jpg)

**Reply by Pierparide** [reply]
Why this error when check IPF version?
![6CFB5993-2608-41FB-AD27-2FE421BA312A](https://user-images.githubusercontent.com/94765448/226097506-ae2b5713-f703-42d0-b2df-de4307efab49.jpeg)

**Reply by Pierparide** [reply]
Why this when visualize dem?

**Reply by Pierparide** [reply]
![D13FE7AD-06F9-4526-A760-14CDADC35457](https://user-images.githubusercontent.com/94765448/226097541-7f09ee68-4fb3-4b7e-8182-ca8ed8612b50.jpeg)

**Reply by Pierparide** [reply]
The dem tiles are detector but processing progress bar stop at 28.6%
![image](https://user-images.githubusercontent.com/94765448/226097652-9720b42a-0622-412d-8fb2-b9221d52d6c7.jpg)

**Reply by blueyanberry** [reply]
Hi, I saw check IPF error too. So, I just download SLCs from ASF and drop in */slc instead of download via Ez-insar. It's works for me!

**Comment by alexisInSAR** [comment]
Hi, 

It seems that you have an issue with the creation of your DEM. At this moment, I can’t understand your problem due to the lack of information. 

However, my advice is to download the DEM using the ISCE tool: 
	1) Open a terminal
	2) In your home directory (e.g., /home/[your username]/) create a file named .netrc 
		cd /home/shami
		nano .netrc  
————————
machine [urs.earthdata.nasa.org](https://emea01.safelinks.protection.outlook.com/?url=http%3A%2F%2Furs.earthdata.nasa.org%2F&data=05%7C01%7C%7C066942522c044799ef0508db131bee9e%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638124782937149932%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=%2BY6tycQMVC3w1%2BSrLxTggyAsEs8fnGZenVs6Hv6SAPI%3D&reserved=0)
	      login [your Earthdata login]
	      password [your Earthdata password]
———————— 					
	3) Go to your work directory
	4) Delete the dem directory and DEM_files.txt
	5) Create a new dem directory (i.e., mkdir dem) and cd dem
	6) Run the following command: 

dem.py -a stitch -b [your coordinates] -s 1 -c -u [http://step.esa.int/auxdata/dem/SRTMGL1](https://emea01.safelinks.protection.outlook.com/?url=http%3A%2F%2Fstep.esa.int%2Fauxdata%2Fdem%2FSRTMGL1&data=05%7C01%7C%7C066942522c044799ef0508db131bee9e%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638124782937149932%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=QBGyGUWuHcBkqjXZHYVrA6MxVthF%2Fwqtk7okvFY44qI%3D&reserved=0) 

7) In EZ-InSAR, after clicking on Select the DEM, select Localhosted DEM and open the dem directory. 

If it does not work, your issue should be with ISCE2 (and potentially GDAL).

**Reply by Pierparide** [reply]
The last command is a python script or in terminal? Coordinate format is right?

**Comment by Pierparide** [comment]
![image](https://user-images.githubusercontent.com/94765448/227730664-4eba1763-370a-42cd-9e38-b87e3331e4ea.jpg)

**Reply by alexisInSAR** [reply]
I can see three problems: 
1. EZ-InSAR does not currently run on MacOS, only on Linux. Some parts may work but not the full InSAR processing. This command may work on MacOS because it is a command from ISCE. 
2. The coordinates should be South North West East without []. (see https://github.com/isce-framework/isce2/tree/main/contrib/stack/topsStack) 
3. Before executing a command and starting MATLAB, please load the correct conda environment via "load_insar" command in your Linux terminal.

**Reply by alexisInSAR** [reply]
I think I fixed this issue: see the new version of _isceprocessing.m_. You can download the new script.

**Reply by Pierparide** [reply]
How download it?

**Reply by alexisInSAR** [reply]
https://github.com/alexisInSAR/EZ-InSAR/blob/Version_2_0_1_Beta/EZINSAR_BIN/ISCE_functions/isceprocessing.m

**Reply by Pierparide** [reply]
The dem files now are detected but when put “visualize DEM” the processing in progress bar stop at 28.6%.
In attached pictures 
![image](https://user-images.githubusercontent.com/94765448/228018967-427720f3-d346-4604-b3f8-f18d6631aca0.jpg)

**Comment by Pierparide** [comment]
I thing that problem is because lack dem.wgs84.vrt.xml ad attached snapshot 
![image](https://user-images.githubusercontent.com/94765448/228023387-538b807f-c4f4-4cd5-8fc9-fa2910dddf26.jpg)

**Reply by alexisInSAR** [reply]
It seems that the error come from the run_download_DEM.sh function. Can you try this command alone?
